src/web/nextui/src/app/eval/ResultsCharts.tsx

```diff
@@ -415,13 +426,269 @@ function MetricChart({ table }: ChartProps) {
 }
 
-function ResultsCharts({ columnVisibility }: ResultsChartsProps) {
+function PerformanceOverTimeChart({ table, evalId, config, datasetId }: ChartProps) {
+  const lineCanvasRef = useRef(null);
+  const lineChartInstance = useRef<Chart | null>(null);
+  const [dataset, setDataset] = useState<TestCasesWithMetadata>();
+
+  interface PointMetadata {
+    evalId: string;
+    highlight: boolean;
+    label: string;
+    date: string;
+    score: string;
+  }
+
+  interface Point {
+    x: number;
+    y: number;
+    metadata: PointMetadata;
+  }
+
+  const fetchDataset = async (id: string) => {
+    try {
+      const res = await fetch(`${await getApiBaseUrl()}/api/evals/${id}`);
+      return await res.json();
+    } catch (err) {
+      const error = err as Error;
+      throw new Error(`Fetch dataset data using given id failed: ${error.message}.`);
+    }
+  };
+
+  function ordinalSuffixOf(i: number): string {
+    const j = i % 10,
+      k = i % 100;
+    if (j === 1 && k !== 11) {
+      return i + 'st';
+    }
+    if (j === 2 && k !== 12) {
+      return i + 'nd';
+    }
+    if (j === 3 && k !== 13) {
+      return i + 'rd';
+    }
+    return i + 'th';
+  }
+
+  useEffect(() => {
+    if (!datasetId) return;
+    (async () => {
+      const data = await fetchDataset(datasetId);
+      setDataset(data.data[0]);
+    })();
+  }, [datasetId]);
+
+  useEffect(() => {
+    if (
+      !lineCanvasRef.current ||
+      !evalId ||
+      !datasetId ||
+      !dataset ||
+      dataset.prompts.length <= 1
+    ) {
+      return;
+    }
+
+    if (lineChartInstance.current) {
+      lineChartInstance.current.destroy();
+    }
+
+    const evalIdToIndex = new Map<string, number>();
+    const highestScoreMap = new Map<string, Point>();
+    let currentIndex = 1;
+    const allPoints: Point[] = [];
+
+    dataset.prompts.forEach((promptObject, index) => {
+      const evalIdKey = promptObject.evalId;
+      const isCurrentEval = evalIdKey === evalId;
+
+      if (!evalIdToIndex.has(evalIdKey)) {
+        evalIdToIndex.set(evalIdKey, currentIndex);
+        currentIndex++;
+      }
+
+      if (promptObject.prompt.metrics) {
+        const point: Point = {
+          x: evalIdToIndex.get(evalIdKey)!,
+          y:
+            (promptObject.prompt.metrics.testPassCount /
+              (promptObject.prompt.metrics.testPassCount +
+                promptObject.prompt.metrics.testFailCount)) *
+            100,
+          metadata: {
+            evalId: promptObject.evalId,
+            highlight: isCurrentEval,
+            label: promptObject.prompt.label,
+            date: promptObject.evalId.split('T')[0],
+            score: promptObject.prompt.metrics.score.toFixed(2),
+          },
+        };
+
+        allPoints.push(point);
+
+        if (!highestScoreMap.has(evalIdKey) || highestScoreMap.get(evalIdKey)!.y < point.y) {
+          highestScoreMap.set(evalIdKey, point);
+        }
+      }
+    });
+
+    if (evalIdToIndex.size == 1) {
```

Comment:
I'm not actually able to get this chart to show as it always hits this condition.  Elsewhere you are restricting this chart to a single prompt and a single dataset.  I might be missing something here but that seems like why?